src/internal/helm/destroy.go

```diff
@@ -57,5 +58,21 @@ func Destroy(purgeAllZarfInstallations bool) {
 		}
 	}
 
+	spinner.Updatef("Checking namespaces for zarf metadata")
+	if namespaces, err := k8s.GetNamespaces(); err != nil {
+		message.Error(err, "Unable to get k8s namespaces")
+	} else {
+		for _, namespace := range namespaces.Items {
+			if _, ok := namespace.Labels["zarf.dev/agent"]; ok {
+				spinner.Updatef("Removing Zarf Agent label for namespace %v", namespace.Name)
+				delete(namespace.Labels, "zarf.dev/agent")
+				if _, err = k8s.UpdateNamespace(&namespace); err != nil {
```

Comment:
If `purgeAllZarfInstallations` is provided, does it make more sense to delete the namespace instead of just removing the label?

Reply:
No this is a separate concept, we already delete namespaces, this is for mutating namespaces that existed outside of Zarf (prior to `zarf init`) that we need to put back.

Reply:
I did move the function though to a better location.